packages/webamp/js/components/FFTNullsoft.ts

```diff
@@ -0,0 +1,243 @@
+// The Web Audio API's FFT is bad, so this exists now!
+// Taken from https://github.com/WACUP/vis_classic/tree/master/FFTNullsoft
+
+export class FFT {
+    private m_samples_in: number;
+    private NFREQ: number;
+    private bitrevtable: number[] | null = null;
+    private envelope: Float32Array | null = null;
+    private equalize: Float32Array | null = null;
+    private temp1: Float32Array | null = null;
+    private temp2: Float32Array | null = null;
+    private cossintable: Float32Array[] | null = null;
+
+    constructor() {
+        this.m_samples_in = 0;
+        this.NFREQ = 0;
+    }
+
+    public init(samples_in: number, samples_out: number, bEqualize = 1, envelope_power = 1.0, mode = false): void {
```

Comment:
Would it be possible to move this code into the constructor? Looks like it's only ever called as part of creating the class instance?

Reply:
I did notice that jberg had his own version of FFTNullsoft for butterchurn, so I'll take some inspiration from it.

Reply:
Managed (or at least attempted to) address this.